packages/esm-ohri-pmtct-app/src/form-entry/post-submission-actions/mother-child-linkage-action.ts

```diff
@@ -87,6 +96,24 @@ async function constructPatientObjectFromObsData(obsGroup, encounterLocation: st
   }
   return null;
 }
+async function updatePatientPtracker(encounter, encounterLocation, patientUuid) {
+  const inComingPTrackerID = encounter.obs.find((observation) => observation.concept.uuid === pTrackerIdConcept).value;
+
+  const patientIdentifiers = await fetchPatientIdentifiers(patientUuid);
+  const exixtingPTrackers = patientIdentifiers.filter((id) => id.identifierType.uuid === PTrackerIdentifierType);
+  if (exixtingPTrackers.some((ptracker) => ptracker.identifier === inComingPTrackerID)) {
+    return;
+  }
+
+  //add current ptracker to identities
+  const currentPTrackerObject: PatientIdentifier = {
+    identifier: inComingPTrackerID,
+    identifierType: PTrackerIdentifierType,
+    location: encounterLocation,
+    preferred: false,
+  };
+  saveIdentifier(currentPTrackerObject, patientUuid);
+}
```

Comment:
Can you re-use this code for both submission handlers? You have code duplication.

Reply:
Yah, hadn't rebased on latest dev but have pushed a commit

Reply:
ok